spacy/vectors.pyx

```diff
@@ -52,7 +52,7 @@ cdef class Vectors:
     cdef public object name
     cdef public object data
     cdef public object key2row
-    cdef cppset[int] _unset
+    cdef public cppset[int] _unset
```

Comment:
@honnibal The current master contains `cdef public object _unset`. Was it intended to be made public?

Reply:
@skrcode The considerations are different when it's a normal Python object. On master the value is  a Python `set`, so there's no harm in giving Python access to it --- and we do normally default to being permissive in access, to match Python idioms.

This might be less confusing if you have a look at the C++ being generated by Cython. You can do that by running `cd spacy && cython -a vectors.pyx`, and then opening `vectors.html` in your browser. You can then click the '+' to see how lines have been compiled.

If we access `vectors._unset` from Python, a `PyObject*` needs to be created. Python can't deal with the `cppset` container directly. Cython is smart enough to translate the `cppset` into a Python `set`, but that's actually not that clever in our situation. The implicit conversion would mean that the following two pieces of code would do very different things:

```python
# values will be a Python set.
def get_unset1(vectors):
    values = vectors._unset
# Values will be a cppset. 
def get_unset2(Vectors vectors):
    values = vectors._unset
```

If we make `._unset` public, and we access it from code where Cython doesn't assign a type to our instance, Cython will happily ccompile the code into a Python-level access to the variable --- which will require making a new `PyObject*` object to return (specifically, a Python `set` object).  If we *don't* make `._unset` public, we'll get an `AttributeError` (unfortunately at runtime...) when this happens, which is at least better than the incorrect behaviour.

Again, the calculation is different if `vectors._unset` is an actual Python object. If it's a Python object, and we make an untyped access, Cython can just return the actual `PyObject*`.

So when the value of `vectors._unset` was a Python `set`, it was safe to make it `public`. When the value is a `cppset`, it's not safe to make it public.